src/Compilers/CSharp/Test/Emit3/Attributes/AttributeTests_Embedded.cs

```diff
@@ -193,51 +198,312 @@ public static void Main()
             CompileAndVerify(code, parseOptions: TestOptions.Regular.WithNoRefSafetyRulesAttribute(), verify: Verification.Passes, expectedOutput: "3");
         }
 
+        [Theory]
+        [InlineData("internal")]
+        [InlineData("public")]
+        public void EmbeddedAttributeFromSourceValidation_Valid(string ctorAccessModifier)
```

Comment:
It looks like two source generator could easily end up in a situation when they define conflicting type in the same compilation. Even if the following partial declaration is used, there will be an error about a duplicate ```[Embedded]``` attribute.
```
namespace Microsoft.CodeAnalysis
{
    [Embedded]
    internal sealed partial class EmbeddedAttribute : System.Attribute
    {
    }
}
```

If the following declaration is used, there will be an error about duplicate `AttributeUsage` attribute.
```
namespace Microsoft.CodeAnalysis
{
    [Embedded]
    [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
    internal sealed partial class EmbeddedAttribute : System.Attribute
    {
    }
}
```

I think we should find a way for source generators to avoid a conflict like this.
For example, instead of reporting an error when `[Embedded]` is not applied, compiler could synthesize its application on a user defined type.

Reply:
If we are planning VB source generators to rely on the same attribute, I think we should also test how similar situation can be handled in VB.

Reply:
> It looks like two source generator could easily end up in a situation when they define conflicting type in the same compilation. 

Indeed. I expect the next PR to add a public API for a source generator to register that they would like the attribute to be generated; we can then take care of doing that generation, only adding the source once. I think that's separable though, and this internal functionality should go in ahead of an api review meeting to add that api.

Reply:
I think in order to continue we need to agree on the direction first. In my personal opinion a solution that doesn't require any special coordination between the parties involved (use of special API etc.) would be preferred. Especially when an explicit user's code itself can participate without any knowledge in advance.

Reply:
I'd prefer an explicit API over an implicit API; automatically synthesizing an attribute definition when a lookup fails in source is an implicit API the compiler would be providing. I also don't like the types of downstream impacts that could have:

1. We'd expose a synthesized type that is applied to source. I don't know of another location we do that today, and it will likely have a number of downstream impacts on our API consumers. What would the IDE show for Go-To-Definition, for example?
2. Where does this attribute show up in our general APIs? Will it be returned from `GetTypeByMetadataName`, even if it was not applied to any source tree, or was applied to a source tree that has not yet been bound? Will it appear in the global namespace if I try to iterate all symbols defined in this compilation? Will it appear in the merged global namespace across all assemblies?

Having an explicit definition in a file, whether that's the user defining it, or the generator driver providing an API to define it if it hasn't been defined yet, is, in my opinion, the more logically and consistent approach, with less potential impact and implementation challenges. However, regardless of that, I think this PR is not dependent on a firm decision on what public API should be provided, and should be taken regardless. Our existing behavior around `EmbeddedAttribute` is inconsistent and should be cleaned up, even if we do not make any further changes in this area.

Reply:
Clarified offline with @AlekseyTs. I misunderstood his original idea, and I like the approach. I'll take a look at that this afternoon.

Reply:
I think we should have a test confirming that the following code can be compiled successfully:
```
namespace Microsoft.CodeAnalysis
{
    internal sealed partial class EmbeddedAttribute : System.Attribute
    {
    }
}

namespace Microsoft.CodeAnalysis
{
    internal sealed partial class EmbeddedAttribute : System.Attribute
    {
    }
}
```